src/logs/metadata_engine/jmespath.py

```diff
@@ -44,26 +44,96 @@ def _func_if(self, condition, if_true_expression, if_false_expression, node_scop
         else:
             return if_false_expression.visit(if_false_expression.expression, node_scope)
 
-    @functions.signature({"types": ['string']},
-                         {"types": ['array-string']})
+
+    @functions.signature({'types': ['string', 'null']}, {'types': ['string', 'null']})
+    def _func_starts_with(self, search, suffix):
+        if(search is None or suffix is None):
+            return False
+        return search.startswith(suffix)
+
+    @functions.signature({'types': ['string', 'null']},
+                         {"types": ['array']})
     def _func_format(self, pattern, values):
-        assert pattern.count("{}") == len(values)
+        return self.format(pattern, values)
 
-        output = pattern
+    def format(self, pattern, values):
+        if pattern.count("{}") != len(values):
+            return None
 
+        output = pattern
         for value in values:
-            output = output.replace("{}", value, 1)
-
+            if value is not None:
+                output = output.replace("{}", value, 1)
         return output
 
-    @functions.signature({'types': ['string']},
-                         {'types': ['string']})
+    @functions.signature({'types': ['string', 'null']},
+                         {"types": ['array']})
+    def _func_format_arn(self, pattern, values):
```

Comment:
Agreed, this may come in handy in other places and will be easier to spot by a person needing it in the future if we name it more explicitly (for example, like Marek recommended)

Reply:
I understand your reasoning, it's all good! I just feel like trying to convey such additional meaning through function name gets a little confusing. If I see arn/id field in config with function "format_required" or "format_require_nonnulls" or whatever then I might realise this reasoning easier